v1/common/amqp.go

```diff
@@ -111,6 +131,54 @@ func (ac *AMQPConnector) Open(url string, tlsConfig *tls.Config) (*amqp.Connecti
 	return conn, channel, nil
 }
 
+func (ac *AMQPConnector) getConn(url string, tlsConfig *tls.Config, keepAlive bool) (*amqp.Connection, error) {
+	ac.mu.Lock()
+	defer ac.mu.Unlock()
+
+	if !keepAlive {
+		return ac.createNewConn(url, tlsConfig)
+	}
+
+	var done, makeNew bool
+	for !done {
```

Comment:
Maybe I'm not getting some of the intricacies of this loop with select. So maybe it's needed, but from what I can see I get these questions:
- Why is this a loop? Why is it not just a non blocking select? Do you need to consume more than 1 message from the `connChan`?
- It doesn't matter for the end result if `connChan` returns a `nil` or an `error`, is this on purpose? If it's an `error`, `makeNew` is set to true and a new loop happens, which sets `done` to true. If it's `nil` this is done in one step. Both options result in `done` and `makeNew` being true.

Reply:
1) it's a loop because there could potentially be multiple messages on the error chan, but your next comment will make this change
2) it's on purpose, because if the error is not nil it would have made sense to process potential remnants in the channel, but then again your comment will make this change

Reply:
Ah ok, makes sense. But yeah both are also fixed by creating a new channel for each new connection.